test/Microsoft.AspNetCore.Mvc.FunctionalTests/ApplicationModelTest.cs

```diff
@@ -166,5 +166,67 @@ public ApplicationModelTest(MvcTestFixture<ApplicationModelWebSite.Startup> fixt
             // Assert
             Assert.Equal("/Home/CannotBeRouted", response.Headers.Location.ToString());
         }
+
+        [Fact]
+        public async Task ProductsArea()
```

Comment:
Can you give these some better names? `ApplicationModel_CanDuplicateController_InMultipleAreas`. Also, this is a pretty good case for `[Theory]`.